File: DelDirInfo/DelDirInfo.go
// DelDirInfo
package DaeseongLib

import (
	_ "fmt"
	"io/ioutil"
	"os"
	"path/filepath"
	"sort"
)

var (
	allfileList = []string{}
)

func FindFiles(sPath string) {

	files, err := ioutil.ReadDir(sPath)
	if err != nil {
		panic(err)
	}

	for _, f := range files {

		sDir := filepath.Join(sPath, f.Name())
		if f.IsDir() {
			FindFiles(sDir)
		} else {
			allfileList = append(allfileList, sDir)
		}
	}
}

func FindFileList(sPath string) []string {
	fileList := []string{}
	filepath.Walk(sPath, func(p string, f os.FileInfo, err error) error {
		if err != nil {
			return nil
		}
		if !f.IsDir() {
			fileList = append(fileList, p)
		}
		return nil
	})
	sort.Strings(fileList)
	return fileList
}

func RemoveDir(sPath string) bool {

	err := os.RemoveAll(sPath)
	if err != nil {
		return false
	}
	return true
}

func FindDirList(sPath string) []string {
	fileList := []string{}

	filepath.Walk(sPath, func(p string, f os.FileInfo, err error) error {
		if err != nil {
			return nil
		}
		if f.IsDir() {
			fileList = append(fileList, p)
		}
		return nil
	})
	return fileList
}

/*
func f1() {
	FindFiles("C:\\test")
	for _, f := range allfileList {
		fmt.Println(f)
	}
}

func f2() {
	fileList := FindFileList("C:\\test")
	for _, f := range fileList {
		fmt.Println(f)
	}
}

func f3() {
	RemoveDir("C:\\test")
}

func f4() {
	DirList := FindDirList("C:\\test")
	for _, dir := range DirList {
		RemoveDir(dir)
	}
}

func main() {
	//f1()
	//f2()
	//f4()
}
*/

File: DelDirInfo/DelDirInfo/DelDirInfo.py
import os
import shutil


def isFile(path):
    if not os.path.exists(path):
        return False
    else:
        return True


def FindFileList(path):
    fileList = []
    for root, subFolders, files in os.walk(path):
        for f in files:
            if isFile(root + os.sep + f):
                fileList.append(root + os.sep + f)
    return fileList


def FindDirList(path):
    dirList = []
    for root, subFolders, files in os.walk(path, topdown=False):
        dirList.append(root)
    return dirList


if __name__ == '__main__':

    # 파일, 디렉토리 모두 삭제
    # shutil.rmtree('E:\\test')

    # 파일만 먼저 지우고
    FileList = FindFileList('E:\\test')
    for i in range(len(FileList)):
        # print(FileList[i])
        os.remove(FileList[i])

    # 파일 삭제 완료후 제일 하위 폴더 부터 차례로 삭제
    dirList = FindDirList('E:\\test')
    for dir in dirList:
        # print(dir)
        os.rmdir(dir)

    pass
